contrib/Shopify/sarama/sarama.go

```diff
@@ -128,7 +128,7 @@ func (p *syncProducer) SendMessages(msgs []*sarama.ProducerMessage) error {
 	}
 	err := p.SyncProducer.SendMessages(msgs)
 	for i, span := range spans {
-		finishProducerSpan(span, msgs[i].Partition, msgs[i].Offset, err)
+		finishProducerSpan(span, msgs[i], err)
```

Comment:
I'm not sure sarama's SyncProducer modify original message with the valid Partition, Offset .